base/libgit2/oid.jl

```diff
@@ -22,18 +19,31 @@ end
 function GitHash(id::AbstractString)
     bstr = String(id)
     len = sizeof(bstr)
-    oid_ptr  = Ref(GitHash())
-    err = if len < OID_HEXSZ
-        ccall((:git_oid_fromstrn, :libgit2), Cint,
-              (Ptr{GitHash}, Ptr{UInt8}, Csize_t), oid_ptr, bstr, len)
-    else
-        ccall((:git_oid_fromstrp, :libgit2), Cint,
-              (Ptr{GitHash}, Cstring), oid_ptr, bstr)
+    if len < OID_HEXSZ
+        throw(ArgumentError("Input string is too short, use `GitShortHash` for partial hashes"))
     end
-    err != 0 && return GitHash()
+    oid_ptr = Ref{GitHash}()
+    @check ccall((:git_oid_fromstrn, :libgit2), Cint,
```

Comment:
Yes, `fromstrp` is just the null-terminated version of `fromstrn`. Since Julia strings store the length, we might as well use `fromstrn`.